packages/components/src/popover/index.js

```diff
@@ -361,38 +216,122 @@ const Popover = ( {
-		anchorRect,
-		getAnchorRect,
-		anchorRef,
-		shouldAnchorIncludePadding,
-		anchorVerticalBuffer,
-		anchorHorizontalBuffer
-	);
+	useEffect( () => {
+		const resize = () => setIsMobileViewport( window.innerWidth < 782 );
+		window.addEventListener( 'resize', resize );
+		return () => window.removeEventListener( 'resize', resize );
+	} );
 
 	// Content size
 	const contentSize = useInitialContentSize( contentRef );
+
 	useEffect( () => {
-		if ( contentSize ) {
-			setIsReadyToAnimate( true );
+		if ( isMobileViewport && expandOnMobile ) {
+			return;
+		}
+
+		if ( ! contentSize ) {
+			return;
 		}
-	}, [ contentSize ] );
 
-	// Compute the position
-	const popoverPosition = usePopoverPosition(
-		anchor,
+		const refresh = () => {
+			let anchor = computeAnchorRect(
+				anchorRefFallback,
+				anchorRect,
+				getAnchorRect,
+				anchorRef,
+				shouldAnchorIncludePadding
+			);
+
+			if ( ! anchor ) {
+				return;
+			}
+
+			anchor = addBuffer(
+				anchor,
+				anchorVerticalBuffer,
+				anchorHorizontalBuffer
+			);
+
+			const {
+				popoverTop,
+				popoverLeft,
+				xAxis,
+				yAxis,
+				contentHeight,
+				contentWidth,
+			} = computePopoverPosition( anchor, contentSize, position );
+
+			// Compute the animation position
+			const yAxisMapping = {
+				top: 'bottom',
+				bottom: 'top',
+			};
+			const xAxisMapping = {
+				left: 'right',
+				right: 'left',
+			};
+			const animateYAxis = yAxisMapping[ yAxis ] || 'middle';
+			const animateXAxis = xAxisMapping[ xAxis ] || 'center';
+
+			containerRef.current.setAttribute( 'data-x-axis', xAxis );
+			containerRef.current.setAttribute( 'data-y-axis', yAxis );
+			containerRef.current.style.top = popoverTop + 'px';
+			containerRef.current.style.left = popoverLeft + 'px';
+
+			if ( contentHeight ) {
+				contentRef.current.style.maxHeight = contentHeight + 'px';
+			}
+
+			if ( contentWidth ) {
+				contentRef.current.style.maxWidth = contentWidth + 'px';
+			}
+
+			if ( xAxis === 'center' && yAxis === 'middle' ) {
+				contentRef.current.classList.add( 'is-without-arrow' );
+			} else {
+				contentRef.current.classList.remove( 'is-without-arrow' );
+			}
+
+			setAnimateOrigin( animateXAxis + ' ' + animateYAxis );
+		};
+
+		refresh();
+
+		/*
+		 * There are sometimes we need to reposition or resize the popover that
+		 * are not handled by the resize/scroll window events (i.e. CSS changes
+		 * in the layout that changes the position of the anchor).
+		 *
+		 * For these situations, we refresh the popover every 0.5s
+		 */
+		const intervalHandle = window.setInterval( refresh, 500 );
+		window.addEventListener( 'resize', refresh );
+		window.addEventListener( 'scroll', refresh, true );
+
+		return () => {
+			window.clearInterval( intervalHandle );
+			window.removeEventListener( 'resize', refresh );
+			window.removeEventListener( 'scroll', refresh, true );
+		};
+	}, [
+		isMobileViewport,
+		expandOnMobile,
 		contentSize,
+		anchorRefFallback.current,
```

Comment:
I'm not certain that adding refs in dependencies is a good practice as changes to these don't rerender anyway.

Reply:
Ah, you're right. I didn't think that through.